pkg/controllers/user/nodesyncer/nodessyncer.go

```diff
@@ -123,27 +122,32 @@ func (m *NodesSyncer) syncLabels(key string, obj *v3.Node) error {
 
 	shouldUpdate := false
 	// set annotations
-	if !reflect.DeepEqual(node.Annotations, machine.Spec.DesiredNodeAnnotations) && machine.Spec.DesiredNodeAnnotations != nil {
-		node.Annotations = machine.Spec.DesiredNodeAnnotations
+	if obj.Spec.DesiredNodeAnnotations != nil && !reflect.DeepEqual(node.Annotations, obj.Spec.DesiredNodeAnnotations) {
```

Comment:
not nil check first, otherwise panic

Reply:
This shouldn't matter, right? DeepEqual also compares nil object.

Reply:
@StrongMonkey I saw the panic, it can be either node is nil, or spec is nil. I bet on the former

Reply:
@alena1108 yeah, changing the order should be fine though. I added not nil check for node recently.